stdlib/src/builtin/sort.mojo

```diff
@@ -282,45 +281,40 @@ fn sort[type: DType](inout buff: Pointer[Scalar[type]], len: Int):
     """
 
     @parameter
-    fn _less_than_equal[ty: AnyTrivialRegType](lhs: ty, rhs: ty) -> Bool:
-        return rebind[Scalar[type]](lhs) <= rebind[Scalar[type]](rhs)
+    fn _less_than_equal(lhs: Scalar[type], rhs: Scalar[type]) -> Bool:
+        return lhs <= rhs
 
-    _quicksort[Scalar[type], _less_than_equal](buff, len)
+    _quicksort_reg[_less_than_equal](buff, len)
 
 
 fn sort(inout list: List[Int]):
```

Comment:
Could you delete this API as well?
It is covered by the generic `sort[type: ComparableCollectionElement, //](inout list: List[type]):`
And I think there should be no downsides if we remove this very specialised one.

Reply:
It's slightly different from the generic `sort` as it uses the `_quicksort_reg` variant which contains optimisations for register types. I'm not sure if that's important or not. Shall we decide that in a later PR?